File: app/src/main/java/net/squanchy/search/view/HeaderViewHolder.java
package net.squanchy.search.view;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

public class HeaderViewHolder extends RecyclerView.ViewHolder {

    public HeaderViewHolder(View itemView) {
        super(itemView);
    }

    public void updateWith(HeaderType headerType) {
        ((TextView) itemView).setText(headerType.headerTextResourceId());
    }

    public void updateWith(CharSequence label) {
        ((TextView) itemView).setText(label);
    }
}

File: app/src/main/java/net/squanchy/search/view/HeaderViewHolder.kt
package net.squanchy.search.view

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

class HeaderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    fun updateWith(headerType: HeaderType) {
        (itemView as TextView).setText(headerType.headerTextResourceId())
    }

    fun updateWith(label: CharSequence) {
        (itemView as TextView).text = label
    }
}
